src/Domain.LinnApps/StockMove/MoveStockService.cs

```diff
@@ -0,0 +1,163 @@
+            string to,
+            int? toLocationId,
+            int? toPalletNumber,
+            DateTime? toStockRotationDate,
+            int userNumber)
+        {
+            if (quantity <= 0)
+            {
+                return new RequisitionProcessResult(false, "You must have a valid quantity.");
+            }
+
+            if (string.IsNullOrEmpty(from) || string.IsNullOrEmpty(to))
+            {
+                return new RequisitionProcessResult(false, "A from and to location must be specified.");
+            }
+
+            if (this.IsKardexLocation(from) && this.IsKardexLocation(to))
+            {
+                return new RequisitionProcessResult(false, "You cannot move between two Kardex locations.");
+            }
+
+            var result = new RequisitionProcessResult();
+            if (!reqNumber.HasValue)
+            {
+                var createResult = this.storesPack.CreateMoveReq(userNumber);
+                if (!createResult.Success)
+                {
+                    throw new CreateReqFailureException("Failed to create move req");
+                }
+
+                result.ReqNumber = createResult.ReqNumber;
+            }
+            else
+            {
+                result.ReqNumber = reqNumber.Value;
+            }
+
+            var req = this.requisitionRepository.FindById(result.ReqNumber);
+
+            if (!fromLocationId.HasValue && !fromPalletNumber.HasValue)
+            {
+                this.GetLocationDetails(from, out fromLocationId, out fromPalletNumber);
+            }
+
+            if (!toLocationId.HasValue && !toPalletNumber.HasValue)
+            {
+                this.GetLocationDetails(to, out toLocationId, out toPalletNumber);
+            }
+
+            if (this.IsKardexLocation(from) || this.IsKardexLocation(to))
+            {
+                throw new MoveInvalidException("Not yet implemented");
+            }
+
+            var nextLineNumber = req.Lines.Count() + 1;
+
+            var checkFromLocation = this.storesPack.CheckStockAtFromLocation(
+                partNumber,
+                quantity,
+                from,
+                fromLocationId,
+                fromPalletNumber,
+                fromStockRotationDate);
+            
+            if (!checkFromLocation.Success)
+            {
+                throw new MoveInvalidException("The required stock doesn't exist at that from location.");
+            }
+
+            var moveResult = this.storesPack.MoveStock(
+                result.ReqNumber,
+                nextLineNumber,
+                partNumber,
+                quantity,
+                fromLocationId,
+                fromPalletNumber,
+                fromStockRotationDate,
+                toLocationId,
+                toPalletNumber,
+                toStockRotationDate,
+             null, // !string.IsNullOrEmpty(fromState) ? fromState : checkFromLocation.State,
+                fromStockPool);
```

Comment:
Commented out? I guess this is waiting the next PR?

Reply:
no. that should be uncommented. thanks.